test/generation.test.ts

```diff
@@ -47,31 +47,49 @@ describe('Generation tests', () => {
   })
 
   it('Generator generates same values in two runs with the same seed', () => {
-    /* We need the test cases list (reporter?) to check this in a way that is not implementation dependent
-
-      const sc1 = fc.scenario()
-        .withGenerator(prng, 1234)
-        .forall('a', fc.integer(-10, 10))
-        .forall('b', fc.integer(-10, 10))
+    const sc1 = fc.scenario()
+      .config(fc.strategy().defaultStrategy().withTestCaseOutput())
+      .withGenerator(prng, 1234)
+      .forall('a', fc.integer(-10, 10))
+      .forall('b', fc.integer(-10, 10))
+      .then(({a, b}) => a+b === b+a)
 
-      const sc2 = fc.scenario()
-        .withGenerator(prng, 1234)
-        .forall('a', fc.integer(-10, 10))
-        .forall('b', fc.integer(-10, 10))
+    const sc2 = fc.scenario()
+      .config(fc.strategy().defaultStrategy().withTestCaseOutput())
+      .withGenerator(prng, 1234)
+      .forall('a', fc.integer(-10, 10))
+      .forall('b', fc.integer(-10, 10))
+      .then(({a, b}) => a+b === b+a)
 
-      expect(sc1.check().testCases).to.eql(sc1.check().testCases)
-    */
+    expect(sc1.check().testCases).to.eql(sc2.check().testCases)
   })
 
   it('Generator generates same values in two runs of the same scenario', () => {
-    /* We need the test cases list (reporter?) to check this in a way that is not implementation dependent
+    const sc1 = fc.scenario()
+      .config(fc.strategy().defaultStrategy().withTestCaseOutput())
+      .withGenerator(prng, 1234)
+      .forall('a', fc.integer(-10, 10))
+      .forall('b', fc.integer(-10, 10))
+      .then(({a, b}) => a+b === b+a)
+
+    expect(sc1.check().testCases).to.eql(sc1.check().testCases)
+  })
 
+  it('Generator generates different test cases for two runs with different seeds', () => {
     const sc1 = fc.scenario()
+      .config(fc.strategy().defaultStrategy().withTestCaseOutput())
       .withGenerator(prng, 1234)
       .forall('a', fc.integer(-10, 10))
       .forall('b', fc.integer(-10, 10))
+      .then(({a, b}) => a+b === b+a)
+
+    const sc2 = fc.scenario()
+      .config(fc.strategy().defaultStrategy().withTestCaseOutput())
+      .withGenerator(prng, 4321)
+      .forall('a', fc.integer(-10, 10))
+      .forall('b', fc.integer(-10, 10))
+      .then(({a, b}) => a+b === b+a)
 
-      expect(sc1.check().testCases).to.eql(sc1.check().testCases)
-    */
+    expect(sc1.check().testCases).to.not.eql(sc2.check().testCases)
```

Comment:
This still doesn't constitute a specification for what we expect from testCases. Let's imagine that seed `1234` produced `[{a: 1000, b: 4000}]` and seed `4321` produced `[{a: 500, b: 100}]`. Both tests would still pass, and it is clearly not doing what we expect it to be doing.

A semantically rich test would check if the contents of `testCases` can be generated by the defined arbitraries. It would also if it doesn't contain more arbitraries than those given. It could also assert that it contains at least one element. And there might be other things that I'm simply forgetting...

Those testes should also not be part of the suite pertaining generation. I would advice to define them in a `testcases.test.ts`.

Reply:
I have pushed new changes containing the tests I believe encompass the behaviour of the test case output